cv19ResSupportV3/V3/Gateways/ResidentGateway.cs

```diff
@@ -169,31 +169,31 @@ public Resident PatchResident(int id, PatchResident command)
         {
             try
             {
+                var residentsTable = _helpRequestsContext.ResidentEntities;
+
                 if (Predicates.IsNotNullAndNotEmpty(command.NhsNumber))
                 {
-                    var matchingResident = _helpRequestsContext.ResidentEntities
-                        .FirstOrDefault(r => r.NhsNumber.Replace(" ", "") == command.NhsNumber.Replace(" ", ""));
+                    var matchingResident = residentsTable.FirstOrDefault(r => r.NhsNumber.Replace(" ", "") == command.NhsNumber.Replace(" ", ""));
```

Comment:
This is only a comment for future consideration, but when you're calling FirstOrDefault or anything else that iterates over a list, it can impact performance if you call methods within the iteration. For example, you could set command.NhsNumber.Replace(" ", "") beforehand so it does it only once.

I don't think it's an issue here though. Our dataset is relatively small and a Replace isn't a slow call.

Reply:
I see, this is good to know. I'll make sure to look into this in the future!